API/Repositories/Implementations/MatchesRepository.cs

```diff
@@ -38,6 +38,13 @@ public async Task<Match> UpdateVerificationStatus(int id, int? verificationStatu
         return existing;
     }
 
+    public async Task<Match?> GetByNameAsync(string name)
```

Comment:
Because this is a search method, we want to allow partial records. This should be renamed to `SearchAsync(string search)` and query for something like `WHERE match.Name LIKE search (ignore case)`. If there's anything else we want to search for (such as by match id), we should handle this here.